src/main/java/frc/robot/systems/AnyBotIntake.java

```diff
@@ -0,0 +1,365 @@
+	 * @return FSM state for the next iteration
+	 */
+	private EveryBotIntakeFSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return EveryBotIntakeFSMState.IDLE_STOP;
+		}
+		switch (currentState) {
+			case INTAKING:
+				if (input.isThrottleForward() != forward || prevOuttaking || needsReset) {
+					timer.reset();
+					timer.start();
+					needsReset = false;
+					prevOuttaking = false;
+				}
+				if (timer.hasElapsed(TIME_RESET_CURRENT)) {
+					currLogs[tick % AVERAGE_SIZE] = spinnerMotor.getOutputCurrent();
+					tick++;
+
+					double avgcone = 0;
+					for (int i = 0; i < AVERAGE_SIZE; i++) {
+						avgcone += currLogs[i];
+					}
+					avgcone /= AVERAGE_SIZE;
+					if (avgcone > CURRENT_THRESHOLD) {
+						holding = true;
+						return EveryBotIntakeFSMState.IDLE_STOP;
+					} else {
+						holding = false;
+					}
+				}
+				forward = input.isThrottleForward();
+				if (!input.isIntakeButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+				return EveryBotIntakeFSMState.INTAKING;
+			case IDLE_STOP:
+				if (input.isOuttakeButtonPressed()) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else if (input.isIntakeButtonPressed()) {
+					if (holding) {
+						return EveryBotIntakeFSMState.IDLE_STOP;
+					} else {
+						return EveryBotIntakeFSMState.INTAKING;
+					}
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			case OUTTAKING:
+				if (input.isOuttakeButtonPressed()) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else {
+					prevOuttaking = true;
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+	/* ------------------------ FSM state handlers ------------------------ */
+	/**
+	 * Handle behavior in states.
+	 *
+	 * @param input
+	 */
+	public void handleIntakingState(TeleopInput input) {
+		if (input != null && input.isThrottleForward()) {
+			spinnerMotor.set(INTAKE_SPEED);
+		} else {
+			spinnerMotor.set(-INTAKE_SPEED);
+		}
+
+	}
+
+	private void handleIdleStopState(TeleopInput input) {
+		spinnerMotor.set(KEEP_SPEED);
+		if (holding) {
+			spinnerMotor.set(HOLDING_SPEED * ((input.isThrottleForward()) ? 1 : -1));
+		}
+	}
+
+	private void handleOuttakingState(TeleopInput input) {
+		if (input == null) {
+			if (!hasTimerStarted) {
+				timer.reset();
+				timer.start();
+				hasTimerStarted = true;
+			}
+		}
+		for (int i = 0; i < AVERAGE_SIZE; i++) {
+			currLogs[i] = 0;
+		}
+		if (input.isThrottleForward()) {
+			spinnerMotor.set(RELEASE_SPEED);
+		} else {
+			spinnerMotor.set(-RELEASE_SPEED);
+		}
+		itemType = ItemType.EMPTY;
+		isMotorAllowed = true;
+		holding = false;
+	}
+
+	/**
+	 * Runs the intake behavior for auto deposit.
+	 *
+	 * @return whether the intake has finished running for auto
+	 */
+	public boolean handleAutoOuttakingState() {
+		if (!autoOuttakingTimerStarted) {
+			autoOuttakingTimerStarted = true;
```

Comment:
I don't think you need autoOuttakingTimerStarted, you can just check if the timer value is equal to 0. Also if you need a new timer make a new timer, no need to share a timer with the other code.